packages/api/amplify_api/lib/src/method_channel_api.dart

```diff
@@ -284,55 +325,125 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
     }
   }
 
-  bool _shouldThrow(int statusCode) {
-    return statusCode < 200 || statusCode > 299;
-  }
-
-  RestResponse _formatRestResponse(Map<String, dynamic> res) {
+  AWSStreamedHttpResponse _formatRestResponse(Map<String, dynamic> res) {
     final statusCode = res['statusCode'] as int;
-    final headers = res['headers'] as Map?;
-    final response = RestResponse(
-      data: res['data'] as Uint8List?,
-      headers: headers?.cast<String, String>(),
-      statusCode: statusCode,
-    );
-    if (_shouldThrow(statusCode)) {
-      throw RestException(response);
-    }
-    return response;
+    // Make type-safe version of response headers.
+    final serializedHeaders = res['headers'] as Map?;
+    final headers = serializedHeaders?.cast<String, String>();
+    final rawResponseBody = res['data'] as Uint8List?;
+
+    return AWSStreamedHttpResponse(
+        statusCode: statusCode,
+        headers: headers,
+        body: Stream.value(rawResponseBody?.toList() ?? []));
```

Comment:
toList is not needed. Uint8List is already a List<int>

Reply:
ty, changed for next rev